src/Chapter2.hs

```diff
@@ -832,7 +859,8 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [a] -> [a]
+rotate rotateLength l = drop rotateLength $ take (length l + rotateLength) (cycle l)
```

Comment:
Unfortunately, `cycle` fails in runtime on empty lists :recycle: So you need to handle this case separately for this function to work properly.

Also, you should consider how to handle negative `rotateLength` as well 🙂